### ISSUE TITLE
Runners fail to register when IPs/hostnames are reused

### ISSUE THREAD

**Issue by jeverling** [machulav/ec2-github-runner#128]
It seems that Github runners use the hostname as runner `--name` by default. This lead to runner instances not being able to register for us, because AWS was re-using internal IP addresses.
Setting runner `--name` explicitly to a unique value should fix this.

**Comment by karuppiah7890** [comment]
@jeverling I'm curious as to how this happened. Even if AWS was reusing internal IP addresses, and EC2 host names are based on it's private IP address, it can only cause two _different_ EC2 hosts at different points in time to have the same host name. If that is the case, what's the problem here? Does GitHub expect new names every time a runner registers to GitHub and not reuse any old names used by old runners that don't exist anymore?

**Comment by jeverling** [comment]
> Does GitHub expect new names every time a runner registers to GitHub and not reuse any old names used by old runners that don't exist anymore?

Exactly, Github keeps a registry of runners, and refuses registering a runner with a name that has been used before. It might be possible to deregister runners or solve this in other ways, but I think making default names unique isn't a bad solution to this problem.

**Comment by karuppiah7890** [comment]
Unique name is good in general 👍 I'm gonna use your solution for unique name

**Comment by jeverling** [comment]
> Where exactly would you set `--name` parameter in? Can you please share an example?

Hey sorry, I didn't see your question earlier.
Check out the PR here: https://github.com/machulav/ec2-github-runner/pull/127/files
